src/fidesops/service/masking/strategy/masking_strategy_aes_encrypt.py

```diff
@@ -11,28 +17,53 @@
 from fidesops.service.masking.strategy.format_preservation import FormatPreservation
 from fidesops.service.masking.strategy.masking_strategy import MaskingStrategy
 from fidesops.util.encryption.aes_gcm_encryption_scheme import encrypt
-
+from fidesops.util.encryption.hmac_encryption_scheme import hmac_encrypt_return_bytes
+from fidesops.util.encryption.secrets_util import SecretsUtil
 
 AES_ENCRYPT = "aes_encrypt"
 
 
 class AesEncryptionMaskingStrategy(MaskingStrategy):
     def __init__(self, configuration: AesEncryptionMaskingConfiguration):
         self.mode = configuration.mode
-        self.key = configuration.key
-        self.nonce = configuration.nonce
         self.format_preservation = configuration.format_preservation
 
-    def mask(self, value: Optional[str]) -> str:
+    def mask(self, value: Optional[str], privacy_request_id: Optional[str]) -> str:
         if self.mode == AesEncryptionMaskingConfiguration.Mode.GCM:
-            masked: str = encrypt(value, self.key, self.nonce)
+            masking_meta: Set[MaskingSecretMeta] = self._build_masking_secret_meta()
+            key: bytes = SecretsUtil.get_or_generate_secret(
+                privacy_request_id,
+                [meta for meta in masking_meta if meta.secret_type == SecretType.key][
+                    0
```

Comment:
Are we better off returning the `masking_meta` as a dict:
```
{
    "key": ...,
    "key_hmac": ...,
    ...,
}
```
instead of as a list and having to address `[0]` here?

Relying on `[0]` is brittle since it will error if nothing is there for any reason, and if multiple are present we'll have potentially redundant configuration.